### PR TITLE
Added ability to alias revisions by human-readable project/version numbers.

### PR DESCRIPTION
**Pull request by mfurlend**
Note: This uses PHP 5.4 short array syntax. I can convert it back to the long array() syntax if you'd like.

I added the ability to automatically associate the revision timestamps with human-readable version names. It works like this:
* The user specifies an `aliastable` (e.g. db_alias) and an `aliasprefix` (e.g. my_project_v) in config.ini. 
* The `aliastable` has two columns: `rev` and `alias`, and is populated right after the `versiontable` is populated with the initial timestamp. 
* The `rev` column contains the same timestamp as the `versiontable`, and the `alias` column contains the concatenation of the`aliasprefix` and a numerical suffix ("1" for the initial version).

So, if the timestamp is **1441645435** and the `aliasprefix` is **my_project_v** then the contents of `aliastable` will look like this after the initial `./migration.php schema` command:

rev | alias
----- | ------
1441645435 | my_project_v1

Whenever a new timestamp is `INSERT`ed into the `versiontable` it is also `INSERT`ed into the `aliastable`. The numerical "version" suffix of the alias is equal to the maximum of all existing suffixes + 1. (Or simply the `COUNT(*)+1` of the `aliastable`). This ensures that, like the timestamp, the suffix of the alias is monotonically increasing (my_project_v1, my_project_v2, my_project_v3.. etc). 

Unlike the `versiontable`, the `aliastable` is not truncated during migration. This is done to avoid a situation where reversion to an old migration, followed by `./migration.php create` would overwrite the initial suffix-version. 

With this modification you are now able to refer to migrations by their alias when invoking the `./migration.php migrate` command.

Example:
```
furlender@mac ~/mmp $ ./migration.php list
         Mon, 07 Sep 2015 13:56:59 -0400    (my_project_v2)
 ***     Mon, 07 Sep 2015 14:00:54 -0400    (my_project_v3)

furlender@mac ~/mmp $ ./migration.php migrate my_project_v1
Will migrate: Mon, 07 Sep 2015 14:00:54 -0400  (my_project_v3) ---> Mon, 07 Sep 2015 13:55:18 -0400  (my_project_v1)

ROLLBACK: Mon, 07 Sep 2015 14:00:54 -0400  (my_project_v3)
ROLLBACK: Mon, 07 Sep 2015 13:56:59 -0400  (my_project_v2)
NOW AT:   Mon, 07 Sep 2015 13:55:18 -0400  (my_project_v1)

furlender@mac ~/mmp $ ./migration.php migrate my_project_v3
Will migrate: Mon, 07 Sep 2015 13:55:18 -0400  (my_project_v1) ---> Mon, 07 Sep 2015 14:00:54 -0400  (my_project_v3)

APPLY:    Mon, 07 Sep 2015 13:56:59 -0400  (my_project_v2)
APPLY:    Mon, 07 Sep 2015 14:00:54 -0400  (my_project_v3)
NOW AT:   Mon, 07 Sep 2015 14:00:54 -0400  (my_project_v3)
```

Also, I found a bug in `migrateController.class.php`, line 65:

```
 if($migration < $target_migration) break;
```
should be 

```
 if($migration <= $target_migration) break;
```
because otherwise a rollback goes back 1 migration too far when there are 3 or more migrations. 

Let me know what you think!

### CODE REVIEW

**Comment by Arbuzov** [comment]
I beleve it`s useful feature. But coud your change Helper.class.php to backward compatibility?

File: lib/Helper.class.php [hunk 1]
```diff
@@ -2,450 +2,558 @@
...
-    $parsed_args = array('options' => array(), 'command' => array('name' => null, 'args' => array()));
-
-    array_shift($args);
-    $opts = GetOpt::extractLeft($args, self::$config_tpl);
-    if($opts === false)
+    const TAB = '  ';
+    const UP = 'Up';
+    const DOWN = 'Down';
+
+    static protected $config_tpl
```
Comment (Arbuzov) [inline lib/Helper.class.php]: will not work whith php<=5.3
:cry:

**Comment by mfurlend** [comment]
Sure. All we need for backwards compatibility is long array syntax I think.

**Comment by mfurlend** [comment]
Ok - I changed everything to be compatible with php 5.3 using https://github.com/endel/php-code-downgrade/

**Comment by mfurlend** [comment]
Is there a reason (besides lack of time) that you haven't merged my changes in? If there is something you'd like me to do first - I'd probably be happy to do it :)

**Pull request merged**